ludwig/backend/ray.py

```diff
@@ -358,22 +394,35 @@ def run(
         self,
         train_loop_per_worker: Callable,
         config: Dict[str, Any],
-        dataset: Dict[str, Any],
-        data_loader_kwargs: Dict[str, Any],
-        stream_window_size: Dict[str, Union[None, float]],
-        callbacks: List[Any] = [],
-    ) -> Tuple[Dict, TorchCheckpoint]:
+        dataset: Dict[str, Any] = None,
+        data_loader_kwargs: Dict[str, Any] = None,
+        stream_window_size: Dict[str, Union[None, float]] = None,
+        callbacks: List[Any] = None,
+        exception_on_error: bool = True,
+    ) -> Result:
+        dataset_config = None
+        if dataset is not None:
+            data_loader_kwargs = data_loader_kwargs or {}
+            stream_window_size = stream_window_size or {}
+            dataset_config = self._get_dataset_configs(dataset, stream_window_size, data_loader_kwargs)
+
+        callbacks = callbacks or []
+
         trainer_cls, kwargs = get_dist_strategy(self.strategy).get_trainer_cls(self.backend_config)
         trainer = trainer_cls(
             train_loop_per_worker=train_loop_per_worker,
             train_loop_config=config,
             datasets=dataset,
             scaling_config=self.scaling_config,
-            dataset_config=self._get_dataset_configs(dataset, stream_window_size, data_loader_kwargs),
-            run_config=RunConfig(callbacks=callbacks, verbose=0),
+            dataset_config=dataset_config,
+            run_config=RunConfig(callbacks=callbacks, verbose=1),
```

Comment:
I'd revert back to 0, because this creates too much logging when we run hyperopt.